PopulateTools/gobierto
# Implement Site's logo management via AWS S3 adapter

Connects to #109.

### Release notes

Releasing this one depends on https://github.com/PopulateTools/gobierto-dev/issues/170.

### What does this PR do?

This PR introduces a small library to abstract file upload operations, and integrates it into the Admin's Site management Form object to allow attaching a file from the UI and storing the `public_url` returned by the adapter.

The first and only available adapter is AWS S3, which needs to be configured via environment variables, as usual. These environment vars are needed at the time being:

```
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_REGION=eu-west-1
S3_BUCKET_NAME=gobierto-dev
```

Once the S3 bucket is available and the AWS credentials are reachable from the app instance, an upload process will be performed every time the User submits a new file. All files are kept from now in a UUID format file names:

![screen shot 2016-12-16 at 11 50 50](https://cloud.githubusercontent.com/assets/126392/21260613/16dee81c-c387-11e6-8989-29582993156f.jpg)

### How should this be manually tested?

Get into a Site edit view within the Admin namespace, and submit a new logo in the file field. In the next GET request, the new logo should be shown right above the file field is rendered:

![screen shot 2016-12-16 at 12 05 04](https://cloud.githubusercontent.com/assets/126392/21260890/6ce1f6d6-c388-11e6-92a1-14f45629dee0.jpg)
